src/Lykke.Service.MarketMakerReports.AzureRepositories/InventorySnapshotRepository.cs

```diff
@@ -29,28 +72,162 @@ public async Task InsertAsync(InventorySnapshot inventorySnapshot)
             };
 
             await _storage.InsertAsync(entity);
+            await UpdateIndexes(inventorySnapshot, id);
         }
 
-        public async Task<IEnumerable<InventorySnapshot>> GetAsync(DateTime startDate, DateTime endDate)
+        private async Task UpdateIndexes(InventorySnapshot inventorySnapshot, Guid id)
         {
-            var filter = TableQuery.CombineFilters(
-                TableQuery.GenerateFilterCondition(nameof(InventorySnapshotEntity.PartitionKey),
-                    QueryComparisons.GreaterThanOrEqual,
-                    GetPartitionKey(startDate)),
-                TableOperators.And,
-                TableQuery.GenerateFilterCondition(nameof(InventorySnapshotEntity.PartitionKey),
-                    QueryComparisons.LessThanOrEqual,
-                    GetPartitionKey(endDate)));
+            // Update index for current day, it now must point to this the last received snapshot
+            await _indexStorage.InsertOrReplaceAsync(
+                new InventorySnapshotIndexEntity(GetIndexPartitionKey(inventorySnapshot.Timestamp), IndexForLastRowKey)
+                {
+                    ReferencePartitionKey = GetPartitionKey(inventorySnapshot.Timestamp),
+                    ReferenceRowKey = GetRowKey(id)
+                });
+            
+            
+            // If we have index entries for any future date, they are now are invalid, we can just remove them,
+            // as they will be reconstructed if needed
+                
+            var filter = TableQuery.GenerateFilterCondition(nameof(InventorySnapshotIndexEntity.PartitionKey),
```

Comment:
You can lose indexes if earlier event will be received

Reply:
Dealt with the case. Updating the index for current date if the time of the new entry is after the indexed entry.